Tests are fragile and cross-platform incompatible.

As mentioned in https://github.com/pylipp/financeager/issues/11#issuecomment-542212160 , the tests are fragile and/or cross-platform incompatible.

On Windows 10 x64 I had 10 tests failing.
On MacOS Mojave I had 2 tests failing:

1. 
```
======================================================================
FAIL: test_invalid_host (test.test_httprequests.HttpRequestProxyTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/abhibha/misc/financeager/test/test_httprequests.py", line 51, in test_invalid_host
    self.assertIn("Name or service not known", error_message)
AssertionError: 'Name or service not known' not found in "Error sending request: HTTPConnectionPool(host='weird.foodomain.nope', port=80): Max retries exceeded with url: /periods/2000/standard/1 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x107d51e90>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))"
```

Testing on error explanations is generally a bad idea. As seen here, for MacOS, the underlying error explanation is significantly different from most Linux ones: `nodename nor servname provided, or not known`.

Instead, test on the exact error type as much as possible. In this case that'd be `NewConnectionError` thrown by urllib3, which should be the same across platforms.

2.
```
======================================================================
FAIL: test_verbose (test.test_cli.CliLocalServerTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/mock.py", line 1209, in patched
    return func(*args, **keywargs)
  File "/Users/abhibha/misc/financeager/test/test_cli.py", line 150, in test_verbose
    "Loading custom config from {}".format(TEST_CONFIG_FILEPATH)))
AssertionError: False is not true
```
Classic case of new lines / whitespace breaking tests. The actual string printed to **stderr** (at-least on my Mac) is: 
```
DEBUG 2019-10-15 18:58:48,266 financeager.config:48 Loading custom config from /tmp/financeager-test-config

```
Take note of the newline. `string.endswith` obviously fails. Instead, let's use `in`.

--- a/test/test_cli.py
+++ b/test/test_cli.py
@@ -146,8 +146,9 @@ default_category = no-category"""
     @mock.patch("sys.stderr.write")
     def test_verbose(self, mocked_stderr):
         self.cli_run("print --verbose")
-        self.assertTrue(mocked_stderr.call_args_list[0][0][0].endswith(
-            "Loading custom config from {}".format(TEST_CONFIG_FILEPATH)))
+        self.assertIn(
+            "Loading custom config from {}".format(TEST_CONFIG_FILEPATH),
+            mocked_stderr.call_args_list[0][0][0])
 
     @mock.patch("financeager.offline.OFFLINE_FILEPATH",
                 "/tmp/financeager-test-offline.json")

--- a/test/test_httprequests.py
+++ b/test/test_httprequests.py
@@ -48,7 +48,7 @@ class HttpRequestProxyTestCase(unittest.TestCase):
 
         error_message = cm.exception.args[0]
         self.assertTrue(error_message.startswith("Error sending request: "))
-        self.assertIn("Name or service not known", error_message)
+        self.assertIn("NewConnectionError", error_message)
 
 
 if __name__ == "__main__":

